docs/getting-started/docker-playgrounds/flink_cluster_playground.md

```diff
@@ -25,3 +25,183 @@ under the License.
+
+
+# Log output from org.apache.flink.yarn to the console. This is used by the
+# CliFrontend class when using a per-job YARN cluster.
+log4j.logger.org.apache.flink.yarn=INFO, console
+log4j.logger.org.apache.flink.yarn.cli.FlinkYarnSessionCli=INFO, console
+log4j.logger.org.apache.hadoop=INFO, console
+
+log4j.appender.console=org.apache.log4j.ConsoleAppender
+log4j.appender.console.layout=org.apache.log4j.PatternLayout
+log4j.appender.console.layout.ConversionPattern=%d{yyyy-MM-dd HH:mm:ss,SSS} %-5p %-60c %x - %m%n
+
+# suppress the warning that hadoop native libraries are not loaded (irrelevant for the client)
+log4j.logger.org.apache.hadoop.util.NativeCodeLoader=OFF
+
+# suppress the irrelevant (wrong) warnings from the netty channel handler
+log4j.logger.org.apache.flink.shaded.akka.org.jboss.netty.channel.DefaultChannelPipeline=ERROR, file
+{% endhighlight %}
+
+**log4j-console.properties**
+{% highlight properties %}
+# This affects logging for both user code and Flink
+log4j.rootLogger=INFO, console
+
+# Uncomment this if you want to _only_ change Flink's logging
+#log4j.logger.org.apache.flink=INFO
+
+# The following lines keep the log level of common libraries/connectors on
+# log level INFO. The root logger does not override this. You have to manually
+# change the log levels here.
+log4j.logger.akka=INFO
+log4j.logger.org.apache.kafka=INFO
+log4j.logger.org.apache.hadoop=INFO
+log4j.logger.org.apache.zookeeper=INFO
+
+# Log all infos to the console
+log4j.appender.console=org.apache.log4j.ConsoleAppender
+log4j.appender.console.layout=org.apache.log4j.PatternLayout
+log4j.appender.console.layout.ConversionPattern=%d{yyyy-MM-dd HH:mm:ss,SSS} %-5p %-60c %x - %m%n
+
+# Suppress the irrelevant (wrong) warnings from the Netty channel handler
+log4j.logger.org.apache.flink.shaded.akka.org.jboss.netty.channel.DefaultChannelPipeline=ERROR, console
+{% endhighlight %}
+
+### Output
+
+* Check Input in  Kafka Topic
+```
+docker-compose exec kafka kafka-console-consumer.sh --bootstrap-server localhost:9092 --topic input
+```
+
+* Observe Output
+```
+docker-compose logs -f taskmanager
+```
+
+# Operational Plays
+
+### Listing Running Jobs
+
+```
+docker-compose run --no-deps client flink list
+```
+
+### Upgrading a Job
+
+Cancelling with savepoint
+
+```
+docker-compose run --no-deps client flink cancel -s <job-id>
+```
+
+Restore from Savepoint (--error-rate 0)
+
+```
+docker-compose run --no-deps client flink run -s <savepoint-path> -d /opt/flink/examples/streaming/StateMachineExample.jar --error-rate 0.00 --sleep 10 --backend file --checkpoint-dir file:///tmp/flink-checkpoint-directory
+```
+
+Rescale from Savepoint
+```
+docker-compose run --no-deps client flink run -p 3 -s <savepoint-path> -d /opt/flink/examples/streaming/StateMachineExample.jar --error-rate 0.01 --sleep 10 --backend file --checkpoint-dir file:///tmp/flink-checkpoint-directory
```

Comment:
`--error-rate 0.01` I'm guessing this means the job will periodically fail and restart from checkpoint?

Reply:
No, it will only print and "Alert" to STDOUT. No failures.